packages/react/src/components/Storage/File-Uploader/FileUploader.tsx

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { useEffect } from 'react';
+import { FileUploaderProps } from './types';
+import { UploaderButton } from './UploaderButton';
+import { Previewer } from './Previewer';
+import { View, Text } from '../../../primitives';
+import { UploaderDrop } from './UploaderDrop';
+import { useFileUploader } from './hooks/useFileUploader';
+import { translate } from '@aws-amplify/ui';
+
+/**
+ * [📖 Docs](https://ui.docs.amplify.aws/react/connected-components/storage)
+ */
+
+export function FileUploader({
+  acceptedFileTypes,
+  fileNames,
+  isPreviewerVisible,
+  level,
+  components: customComponents = {},
+  maxFiles,
+  maxMultipleSize,
+  maxSize,
+  multiple = true,
+  onChange,
+  onError,
+  onSuccess,
+  path,
+  showPreview = false,
+  variation = 'button',
+}: FileUploaderProps): JSX.Element {
+  const { UploaderDrop = FileUploader.UploaderDrop } = customComponents;
+  const { setShowPreviewer, showPreviewer, files } = useFileUploader();
+  const { getDropEvents, inDropZone, setFiles } = useFileUploader();
+
+  useEffect(() => {
+    setShowPreviewer(isPreviewerVisible);
+  }, [setShowPreviewer, isPreviewerVisible]);
+
+  function onClose() {
+    setShowPreviewer(false);
+  }
```

Comment:
NIT: prefer 
```suggestion
  const onClose = () => {
    setShowPreviewer(false);
  }
```
for consistency with other React codes

Reply:
Same with `onUpload`, and other inline functions

Reply:
I find this pattern unusual. I always thought it makes more sense to calls these functions instead of having this inline functions. I'll change it either way.